### PR TITLE
chore: create a new TypeORM config using @nestjs/typeorm for public schema

### LINKED ISSUE #1827: chore: create a new TypeORM config using @nestjs/typeorm for public schema
## Scope & Context

Scope and context are explained in this main [ticket #1824](https://github.com/twentyhq/twenty/issues/1824)

## Current behavior

Prisma ORM is used for the public schema queries.

## Expected behavior

We're going to drop Prisma ORM step by step, this first step only add the TypeORM configuration for the public schema using `@nestjs/typeorm`.
`public` is the default schema, so we don't need to provide the name into the configuration.

First we need to create a base configuration for TypeORM by creating a `ormconfig.js` file:

```javascript
require("dotenv").config();

module.exports = {
  url: process.env.PG_DATABASE_URL,
  type: 'postgres',
  entities: [__dirname + '/**/*.entity{.ts,.js}'],
  synchronize: false,
  migrationsRun: true,
  migrationsTableName: '_typeorm_migrations',
  migrations: [__dirname + '/migrations/**/*{.ts,.js}'],
  cli: {
    migrationsDir: __dirname + '/migrations'
  },
};
```

After we'll create a new folder inside `src` called `coreV2` and instantiate TypeORM inside it.
`core-v2` folder will be used during the migration from Prisma to TypeORM to avoid conflit, and drop at the end.

```typescript
@Module({
  imports: [
    TypeOrmModule.forRoot(),
    /* ... other modules ... */
  ],
  /* ... other properties ... */
})
export class CoreV2Module {}
```

### PR DESCRIPTION
**Pull request by gitstart-twenty**
Fixes #1827

### CODE REVIEW

**Review by gitstart-twenty — COMMENTED** [review]
Hey @magrinj,
I added the `user.*` files just to test whether the connection to the DB works. Turns out while  using just `ormconfig.js`, typeorm won't automatically read it as it would if we were using `DataSource`, it will throw the error below:

Screenshot 2023-10-26 at 12 59 08

This is why I opted for the switch to `ormconfig.ts` which I then import into `coreV2/core.module.ts`.

I intend to remove all the `user.*` files before we conclude this PR.

Thanks.

File: server/ormconfig.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import dotenv from 'dotenv';
+import { PostgresConnectionOptions } from 'typeorm/driver/postgres/PostgresConnectionOptions';
+dotenv.config();
+
+export default {
+  url: process.env.PG_DATABASE_URL,
+  type: 'postgres',
+  entities: [__dirname + '/src/core/**/*.entity{.ts,.js}'],
```
Comment (magrinj) [inline server/ormconfig.ts:8]: @gitstart-twenty Should be `coreV2` here, and `userv2` should be placed in `coreV2` too

File: server/src/app.module.ts [hunk 2]
```diff
@@ -34,7 +35,7 @@ import { ExceptionFilter } from './filters/exception.filter';
       context: ({ req }) => ({ req }),
       driver: YogaDriver,
       autoSchemaFile: true,
-      include: [CoreModule],
+      include: [CoreV2Module],
```
Comment (magrinj) [inline server/src/app.module.ts:38]: @gitstart-twenty We want to keep CoreModule working during the migration, so here you should keep `CoreModule` and create a new `GraphQLModule` configuration for `CoreV2Module` with a dedicated path.
Here an example with `metadata`:


```typescript
    GraphQLModule.forRoot<YogaDriverConfig>({
      context: ({ req }) => ({ req }),
      driver: YogaDriver,
      autoSchemaFile: true,
      include: [MetadataModule],
      resolvers: { JSON: GraphQLJSON },
      plugins: [],
      path: '/metadata',
    }),
```

Instead of metadata we can put `graphqlV2`

**Review by magrinj — CHANGES_REQUESTED** [review]

**Comment by gitstart-twenty** [comment]
> Hey @gitstart-twenty, I've request some changes, you can put a really small user entity just to make it work, and extend it in another PR to match the user entity from prisma.

On it

File: server/tsconfig.json [hunk 3]
```diff
@@ -22,6 +22,10 @@
     "forceConsistentCasingInFileNames": false,
     "noFallthroughCasesInSwitch": false,
     "resolveJsonModule": true,
-    "typeRoots": ["@types", "node_modules/@types"]
+    "typeRoots": ["@types", "node_modules/@types"],
+    "paths": {
+      "@/*": ["src/core/*"],
```
Comment (magrinj) [inline server/tsconfig.json:27]: @gitstart-twenty Why did you add this ?

Comment (gitstart-twenty) [reply server/tsconfig.json]: @magrinj This has been removed

**Review by magrinj — APPROVED** [review]
Seems fine to me, you can continue on the next tickets based on this one !
Thanks !

**Comment by gitstart-twenty** [comment]
> Seems fine to me, you can continue on the next tickets based on this one ! Thanks !

Great. On it!
